lib/segment/src/vector_storage/memmap_vector_storage.rs

```diff
@@ -37,12 +38,21 @@ pub fn open_memmap_vector_storage(
     path: &Path,
     dim: usize,
     distance: Distance,
+) -> OperationResult<Arc<AtomicRefCell<VectorStorageEnum>>> {
+    open_memmap_vector_storage_with_async_io(path, dim, distance, get_async_scorer())
+}
+
+pub fn open_memmap_vector_storage_with_async_io(
+    path: &Path,
+    dim: usize,
+    distance: Distance,
+    with_async_io: bool,
```

Comment:
What about making `with_async_io` an `Option<bool>` with `.unwrap_or_else(get_async_scorer)`?

Then we can remove the indirect function.

Reply:
in this case would need to refactor all usages. And put None everywhere